src/main/java/io/gravitee/fetcher/git/GitFetcher.java

```diff
@@ -0,0 +1,64 @@
+/**
+ * Copyright (C) 2015 The Gravitee team (http://gravitee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.gravitee.fetcher.git;
+
+import io.gravitee.fetcher.api.Fetcher;
+import io.gravitee.fetcher.api.FetcherException;
+import org.eclipse.jgit.api.Git;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.*;
+
+/**
+ * @author Nicolas GERAUD (nicolas <AT> graviteesource.com)
+ * @author GraviteeSource Team
+ */
+public class GitFetcher implements Fetcher{
+
+    private final Logger LOGGER = LoggerFactory.getLogger(GitFetcher.class);
+    private GitFetcherConfiguration gitFetcherConfiguration;
+
+    public GitFetcher(GitFetcherConfiguration gitFetcherConfiguration) {
+        this.gitFetcherConfiguration = gitFetcherConfiguration;
+    }
+
+    @Override
+    public InputStream fetch() throws FetcherException {
+        File localPath = null;
+        try {
+            localPath = File.createTempFile("GraviteeioTemporaryGitRepository", "");
```

Comment:
why not indeed.

because files are created in the default temp folder, the OS manage when to remove temp dir.